lms/djangoapps/discussion/urls.py

```diff
@@ -13,7 +13,7 @@
     url(r'^(?P<discussion_id>[\w\-.]+)/threads/(?P<thread_id>\w+)$', 'single_thread', name='single_thread'),
     url(r'^(?P<discussion_id>[\w\-.]+)/inline$', 'inline_discussion', name='inline_discussion'),
     url(
-        r'discussion_board_fragment_view$',
+        r'discussion_board_fragment_view',
```

Comment:
@bogdanl As far as I know, Django's url definitions does not affect presense or absense of the querystring. Could you please explain why this change is necessary?

Reply:
Without the $ at the end, any call to the `discussion_board_fragment_view` would match the root URL handler in that urlconf with querystring appended. Actually the underlying problem might be the missing $ in the root handler, or a combination of the two. Like this it will match both URLs correctly though. I will check again to make sure that's the proper fix.

Reply:
If we don't remove the trailing `$`, the resolver won't match `.../discussion_board_fragment_view/` (with a trailing slash); nor it will match `.../discussion_board_fragment_view?format=json`. If we remove it, it will match cover the trailing slash scenario as well and all combinations of trailing slash with or without querystring will work.

Furthermore the `forum_form_discussion` handler needs a more restrictive regex, as at the moment it matches anything that's not matched in the previous handlers, causing unexpected side effects.

Reply:
@bogdanl If that's the practical effect than you're probably right, but theoretically it shoudln't be the case:

* `discussion_board_fragment_view?format=json` - querystring is not matched by the url pattern
* `/discussion_board_fragment_view/` - if I'm not mistaking trailing slashes must be automatically appended via 302 client redirect (there's a Django setting to do so, and I believe it is enabled).

So, probably we were looking at the problem from the wrong angle - if you just add a trailing slash *and* `$` it might just work.